distributed/comm/tcp.py

```diff
@@ -257,12 +257,17 @@ async def write(self, msg, serializers=None, on_error="message"):
         try:
             # trick to enque all frames for writing beforehand
             for each_frame in frames:
-                each_frame_nbytes = nbytes(each_frame)
-                if each_frame_nbytes:
+                if len(each_frame):
```

Comment:
I think we still need `nbytes` to be used based on Ben's observation (or perhaps the `cast`ing happens earlier like in serialization).

My guess is in Ben's run, we are seeing something like an array with shape `(2, 0)`. So `len` is non-zero, which will allow it to enter the `if`, but it is actually empty. So we want to exclude writing it out, which is what this `if` should accomplish

Reply:
Yeah make sense, I have reintroduced `nbytes(each_frame)`. @quasiben does it work now?